pkg/operator/ceph/spec/spec.go

```diff
@@ -145,3 +146,30 @@ func PodLabels(appName, namespace, daemonType, daemonID string) map[string]strin
 	labels[daemonType] = daemonID
 	return labels
 }
+
+// CheckPodMemory verify pod's memory limit is valid
+func CheckPodMemory(resources v1.ResourceRequirements, cephPodMinimumMemory uint64) error {
+	// Ceph related PR: https://github.com/ceph/ceph/pull/26856
+	podMemoryLimit := resources.Limits.Memory()
+	podMemoryRequest := resources.Requests.Memory()
+	errorMessage := `refuse to run the pod with %dmb of ram, provide at least %dmb.`
+
+	// If nothing was provided let's just return
+	// This means no restrictions on pod's resources
+	if podMemoryLimit.IsZero() && podMemoryRequest.IsZero() {
+		return nil
+	}
+
+	// This means LIMIT and REQUEST are either identical or different but still we use LIMIT as a reference
+	if uint64(podMemoryLimit.Value()) < display.MbTob(cephPodMinimumMemory) {
+		return fmt.Errorf(errorMessage, display.BToMb(uint64(podMemoryLimit.Value())), cephPodMinimumMemory)
+	}
+
+	// This means LIMIT < REQUEST
+	// Kubernetes will refuse to schedule that pod however it's still valuable to indicate that user's input was incorrect
+	if uint64(podMemoryLimit.Value()) < display.MbTob(cephPodMinimumMemory) {
+		return fmt.Errorf(errorMessage+"\nUser has specified a pod memory limit below the pod memory request in the cluster CRD.\nRook will create pods that are expected to fail to serve as a more apparent error indicator to the user.", display.BToMb(uint64(podMemoryLimit.Value())), cephPodMinimumMemory)
```

Comment:
Nit: My preference would be to break long string lines up using the `+` operator for better code readability.
```go
this := "a really long " +
    "string\n" +
    "or a string with newlines?"
```